### PR TITLE
Spring Tool Suite: Automate (un)installation + add license

### PR DESCRIPTION
**Pull request by radeksimko**
Shouldn't we rename the cask to `spring-tool-suite` based on the filename?

License source: https://spring.io/tools

> The Spring Tool Suite as well as the Groovy & Grails Tool Suite are built on top of the Eclipse platform, which is provided by the Eclipse Foundation under the Eclipse Public License.

### CODE REVIEW

File: Casks/sts.rb [hunk 1]
```diff
@@ -10,7 +10,19 @@ def self.based_on_eclipse
...
+  caveats <<-EOS.undent
+    #{token} requires Java 6+, you can install the latest Java using
+
+      brew cask install java
+
+    If you pick an installation directory other than ~/springsource
+    when installing #{token}, you will need to uninstall it manually.
+  EOS
+
+  uninstall :delete => [ '~/springsource' ]
```
Comment (vitorgalvao) [inline Casks/sts.rb]: It doesn’t need to be an array; it works as `uninstall :delete => '~/springsource' `.

**Comment by vitorgalvao** [comment]
> Shouldn't we rename the cask to `spring-tool-suite` based on the filename?

Yes.

**Comment by radeksimko** [comment]
@vitorgalvao PR updated.

File: Casks/spring-tool-suite.rb [hunk 2]
```diff
@@ -10,7 +10,16 @@ def self.based_on_eclipse
...
+  installer :script => "Installer - Spring Tool Suite #{version}.RELEASE.app/Contents/MacOS/JavaApplicationStub",
+            :args => [ '-q' ]
+
+  caveats <<-EOS.undent
+    #{token} requires Java 6+, you can install the latest Java using
+
+      brew cask install java
+  EOS
+
+  uninstall :delete => '~/springsource'
```
Comment (vitorgalvao) [inline Casks/spring-tool-suite.rb]: Shouldn’t this be a `zap` instead?

**Comment by vitorgalvao** [comment]
@radeksimko Care to update rebase so we can merge?

**Comment by vitorgalvao** [comment]
Seems to be no longer applicable.

**Pull request closed without merging**